chris_backend/plugininstances/views.py

```diff
@@ -35,6 +35,8 @@ def create(self, request, *args, **kwargs):
         Overriden to remove descriptors from the request that must take their default
         value on creation.
         """
+        if not isinstance(self.request.data, dict) or not getattr(self.request.data, '_mutable', 1):
```

Comment:
The former `isinstance(..., dict)` is sufficient to fix the issue.

What does `getattr(..., '_mutable', ...)` do? Regardless, the attribute name starts with a `_` prefix, indicating it is a private variable.

Reply:
If the `Content-Type` header - `application/json` is not supplied the data passed defaults to url encoded. So internally it's represented as an immutable dict. If you then try to pop from that dict (which is a child of the dict class) it throws an error as it's immutable.

Reply:
The dict is `django.http.QueryDict`. It supports both mutable and immutable depending on the parameter provided during creation. The only way I could differentiate which way it was initialized, short of a try-catch, was by checking the ._mutable property if present and if currently set to False